src/universal/modules/notifications/components/Notifications/Notifications.js

```diff
@@ -16,28 +16,28 @@ import withStyles from 'universal/styles/withStyles';
 import fromGlobalId from 'universal/utils/relay/fromGlobalId';
 import withMutationProps from 'universal/utils/relay/withMutationProps';
 
-const nop = () => {};
-
 const Notifications = (props) => {
   const {
     atmosphere,
     dispatch,
     notifications,
+    submitMutation,
+    onCompleted,
+    onError,
+    submitting,
     styles
   } = props;
 
-  const clearableNotifs = notifications.edges.map(({node: notification}) => notification).filter((n) => !requiresAction(n));
-  const clearAllNotifications = () =>
-    Promise.all(
-      clearableNotifs
-        // eslint-disable-next-line array-callback-return
-        .map(({id}) => {
-          ClearNotificationMutation(atmosphere, fromGlobalId(id).id, nop, nop);
-        })
-    );
+  const clearableNotifs = notifications.edges.filter(({node}) => !requiresAction(node));
+  const clearAllNotifications = () => {
+    submitMutation();
+    clearableNotifs.forEach(({node}) => {
+      ClearNotificationMutation(atmosphere, fromGlobalId(node.id).id, onError, onCompleted);
```

Comment:
@dan-f i know my `withMutationProps` has some poorly named stuff, but here's how i usually use the stuff. `submitMutation` just gives you a prop called submitting that is `true`. I use that to hide the button so old grannies don't double click the button.
Then, `onCompleted` turns that back to `false`.
`onError` populates a field called `error`, so we could have an error message in case the `clear all` button didn't work, but i think that's a little overkill for this view.

Reply:
Hey thanks! I had explored using the `withMutationProps` HOC, but my main concern was that the `submitMutation`/`onError`/`onCompleted` functions, when called concurrently for many notifications, would race in weird ways, and that the `submitting` value could flicker.